src/VirtoCommerce.CatalogModule.Data/Services/AssociationService.cs

```diff
@@ -78,9 +92,56 @@ public async Task SaveChangesAsync(IHasAssociations[] owners)
                 source.Associations.Patch(target.Associations, associationComparer, (sourceAssociation, targetAssociation) => sourceAssociation.Patch(targetAssociation));
 
                 await repository.UnitOfWork.CommitAsync();
+                //Reset cached associations
+                ItemCacheRegion.ExpireProducts(changedEntities.Select(c => c.ItemId).ToArray());
+                AssociationSearchCacheRegion.ExpireRegion();
```

Comment:
Let's extract this code into ClearCache(string[] productIds) method, as it is used three times in the service.

Reply:
what do you mean? need to rename ExpireProducts to ClearCache? I think, it is not a good decision, because all clear-cache methods in CancellableCacheRegion and its descendants are named as "ExpireSomething"